src/format/parse.rs

```diff
@@ -224,6 +225,263 @@ fn parse_rfc3339<'a>(parsed: &mut Parsed, mut s: &'a str) -> ParseResult<(&'a st
+/// accuracy, and representations with a decimal fraction:
+///
+/// |                                   | basic format | extended format |
+/// |-----------------------------------|--------------|-----------------|
+/// | complete representation           | hhmmss       | hh:mm:ss        |
+/// | reduced accuracy: hour and minute | hhmm         | hh:mm           |
+/// | reduced accuracy: hour            | hh           |                 |
+/// | decimal fraction of the second    | hhmmss,ss̲    | hh:mm:ss,ss̲     |
+/// | decimal fraction of the minute    | hhmm,mm̲      | hh:mm,mm̲        |
+/// | decimal fraction of the hour      | hh,hh̲        | hh,hh̲           |
+///
+/// As decimal sign is both `,` and `.` are allowed, with `,` being preferred.  A decimal fraction
+/// must have at least one digit. The standard puts no limit on the number of digits.
+///
+/// Midnight can be represented with both `00:00` (at the start of the day) and `24:00` (at the end
+/// of the calendar day).
+pub(crate) fn parse_iso8601_time<'a>(
+    parsed: &mut Parsed,
+    mut s: &'a str,
+) -> ParseResult<(&'a str, Iso8601Format, bool)> {
+    use Iso8601Format::*;
+
+    macro_rules! try_consume {
+        ($e:expr) => {{
+            let (s_, v) = $e?;
+            s = s_;
+            v
+        }};
+    }
+
+    // We assume the maximum number of digits to be 14 and ignore any digits after that.
+    // 10^14 * 3600 fits nicely within an `i64` without overflowing.
+    const MAX_FRACTION_DIGITS: usize = 14;
+    let mut format = Unknown;
+
+    let hour = try_consume!(scan::number(s, 2, 2));
+    parsed.set_hour(if hour == 24 { 0 } else { hour })?;
+
+    if let Some(&b',' | &b'.') = s.as_bytes().first() {
+        // Minute, second and nanosecond are expressed as a fraction of an hour.
+        s = &s[1..];
+        let digits = s.as_bytes().iter().take_while(|c| c.is_ascii_digit()).count();
+        let (_, fraction) = scan::number(s, 1, MAX_FRACTION_DIGITS)?;
+        s = &s[digits..];
+        let (seconds, nanos) =
+            fraction_to_seconds(fraction, min(digits, MAX_FRACTION_DIGITS), 3600);
+
+        parsed.set_minute(seconds / 60)?;
+        parsed.set_second(seconds % 60)?;
+        parsed.set_nanosecond(nanos)?;
+        return Ok((s, format, hour == 24));
+    }
+
+    let c = s.as_bytes().first().unwrap_or(&b'a');
+    if !(c.is_ascii_digit() || c == &b':') {
+        // Allow reduced accuracy
+        parsed.set_minute(0)?;
+        return Ok((s, format, hour == 24));
+    }
+
+    format = if s.as_bytes().first() == Some(&b':') { Extended } else { Basic };
+    if format == Extended {
+        s = &s[1..];
+    }
+    parsed.set_minute(try_consume!(scan::number(s, 2, 2)))?;
+
+    if let Some(&b',' | &b'.') = s.as_bytes().first() {
+        // Second and nanosecond are expressed as a fraction of a minute.
+        s = &s[1..];
+        let digits = s.as_bytes().iter().take_while(|c| c.is_ascii_digit()).count();
```

Comment:
In case this is passed a very large chunk of data, this should be limited by `take`.

    s.as_bytes().iter().take(MAX_FRACTION_DIGITS).take_while(|c| c.is_ascii_digit()).count();

Reply:
We only parse `MAX_FRACTION_DIGITS`. But I don't see any problem with allowing more digits, and skipping over as many digits as present beyond `MAX_FRACTION_DIGITS`.